### PR TITLE
feat(nutrition): add FREE recommendations + PRO coverage endpoints (PR-1)

### PR DESCRIPTION
**Pull request by Katsiarynakavaleuskaya**
## Summary

- Add `GET /api/v1/nutrition/recommendations` (FREE tier, no auth) — thin adapter over `core.recommendations.get_nutrient_recommendations()` returning WHO-based daily targets (kcal, macros, micros, water, activity)
- Add `POST /api/v1/pro/nutrition/coverage` (PRO tier, `require_pro_tier`) — scores actual nutrient intake against personalized WHO targets with per-nutrient coverage status (deficient/adequate/excess) and overall score
- 35 tests with 100% coverage on all new code (FREE validation, PRO tier guards, contract tests)

## Scope

**IN:**
- Pydantic v2 schemas: `ProfileInput`, `NutrientRecommendationsResponse`, `NutrientCoverageRequest/Response/Item/Summary`
- FREE router: `app/routers/nutrition_recommendations.py` registered in `legacy_app.py`
- PRO router: `app/routers/pro_nutrition_insights.py` registered in `app/routers/pro_registration.py`
- OpenAPI + schema.ts regenerated (deterministic)
- Test file: `tests/test_nutrition_recommendations_api.py`

**OUT:**
- PRO deficiency recommendations endpoint (PR-2)
- PRO micronutrient targets endpoint (PR-2)
- PRO safety validation endpoint (PR-2)
- `goal` parameter for FREE endpoint (deferred to BACKLOG_LEDGER)
- Rate limiting on coverage endpoint (deferred)

## Verification

```
make lint          ✅ clean
make typecheck     ✅ 0 errors / 218 files
make test-fast     ✅ 87/87 pass
pytest tests/test_nutrition_recommendations_api.py ✅ 35/35 pass (100% cov)
pytest tests/test_repo_policy_guards.py ✅ 12/12 pass
pre-commit run --all-files ✅ 16/16 hooks pass
make openapi       ✅ deterministic
```

## Test plan

- [x] FREE endpoint returns valid JSON with kcal_daily > 0 for all 5 activity levels
- [x] FREE endpoint returns 422 for invalid age/gender/weight/height/activity
- [x] Male vs female targets differ for same profile
- [x] Boundary values (age=18, age=120, weight=30kg) accepted
- [x] Pediatric age (< 18) rejected with 422 (adults-only API)
- [x] PRO coverage endpoint returns 401/403 without auth headers
- [x] PRO coverage endpoint returns 200 with `pro_headers` and `vip_headers`
- [x] Coverage status correctly reflects deficient/adequate/excess
- [x] Summary counts add up to total_nutrients
- [x] overall_score in [0, 100]
- [x] Empty consumed dict returns 422
- [x] Negative consumed values rejected at 422
- [x] Content-Type assertions on all JSON responses

## Deferred / Follow-ups

- `calculate_weekly_coverage()` endpoint → VIP tier (BACKLOG_LEDGER)
- `goal` param for FREE endpoint (currently hardcoded "maintain")
- i18n for safety warnings (EN-only initially)
- Rate limiting on coverage endpoint

## Discussion Thread Pass

- [x] Discussion-thread pass completed
- [x] Fixed in commit mapping completed

### Fixed in Commit Mapping

- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#pullrequestreview-3853827762 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#pullrequestreview-3853901439 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#pullrequestreview-3853956426 -> 6e095c52
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#pullrequestreview-3854039093 -> 6e095c52
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852642754 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852642767 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852642772 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852761523 -> 6e095c52
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852711365 -> 70f311fa
- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/899#discussion_r2852837952 -> 6e095c52

## Merge Readiness

- [ ] CI green
- [ ] CodeRabbit / Sourcery / Cubic — no actionables
- [ ] One review cycle after latest bot activity

🤖 Generated with [Qoder][https://qoder.com]

## Summary by CodeRabbit

* **New Features**
  * FREE: Added personalized nutrient recommendations endpoint (daily kcal, macros, micros, water) based on user profile
  * PRO: Added nutrient coverage analysis endpoint returning per-nutrient coverage, statuses, and summary scores

* **Documentation**
  * OpenAPI/docs updated with new request/response schemas and API paths for both FREE and PRO nutrition surfaces

* **Tests**
  * Added comprehensive test suite covering validation, edge cases, auth guards, and coverage scoring logic

### CODE REVIEW

**Comment by Katsiarynakavaleuskaya** [comment]
**Re: Cubic P2 (age=18)** — Intentional change per CodeRabbit review: `ge=18` aligns anthropometric bounds (weight >= 30 kg, height >= 100 cm are adult-only ranges). PR body updated to reflect new boundary (age=18) and added explicit test `test_invalid_age_pediatric_rejected_422`. No action needed.

File: tests/test_nutrition_recommendations_api.py [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        resp = client.post(_PRO_COVERAGE_URL, json=_VALID_COVERAGE_BODY, headers={})
+        assert resp.status_code in (401, 403)
+
+    def test_coverage_pro_key_200(self, client: TestClient, pro_headers: dict[str, str]) -> None:
+        resp = client.post(_PRO_COVERAGE_URL, json=_VALID_COVERAGE_BODY, headers=pro_headers)
+        assert resp.status_code == 200
+
+    def test_coverage_vip_key_200(self, client: TestClient, vip_headers: dict[str, str]) -> None:
+        resp = client.post(_PRO_COVERAGE_URL, json=_VALID_COVERAGE_BODY, headers=vip_headers)
+        assert resp.status_code == 200
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/test_nutrition_recommendations_api.py:213]: Intentional: `test_coverage_vip_key_200` verifies that VIP-tier keys (superset of PRO) are also accepted on PRO endpoints. There is already `test_coverage_pro_key_200` using `pro_headers` for the primary happy path. This test is the backward-compatibility/tier-fallback assertion. No change needed.
